src/Mod/Sketcher/Gui/TaskSketcherConstraints.cpp

```diff
@@ -1224,12 +1224,17 @@ void TaskSketcherConstraints::onListWidgetConstraintsItemChanged(QListWidgetItem
     // otherwise a checkbox change will trigger a rename on the first execution, bloating the
     // constraint icons with the default constraint name "constraint1, constraint2"
     if (newName != currConstraintName && !basename.empty()) {
-        std::string escapedstr = Base::Tools::escapedUnicodeFromUtf8(newName.c_str());
+        if (newName != Base::Tools::getIdentifier(newName)) {
+            Gui::NotifyUserError(
+                sketch, QT_TRANSLATE_NOOP("Notifications", "Value Error"),

```

Comment:
Do you think there may be value in making a helper function that does this check, so as to avoid a mistake if this policy changes in the future?